pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -92,6 +93,15 @@ func (d *ChainDescriber) Describe(ist *imageapi.ImageStreamTag, includeInputImag
 		return "", NotFoundErr(fmt.Sprintf("%q", ist.Name))
 	}
 
+	markers := buildanalysis.FindCircularBuilds(g, osgraph.DefaultNamer)
+	if len(markers) > 0 {
+		for _, marker := range markers {
+			if strings.Contains(marker.Message, ist.Name) {
+				return fmt.Sprintf("Exiting examination of %s. %s", ist.Name, marker.Message), nil
```

Comment:
Using the same yaml, and with the preface that I did not start any builds (not everything in the test config exists), `oc status -v` produces the following (where the cycle message is one of the warnings as @kargakis mentioned):

```
gmontero ~/go/src/github.com/openshift/origin  (issue3902)$ oc status -v
In project example on server https://localhost:8443

bc/ruby-22-centos7 docker build of https://github.com/openshift/ruby-hello-world
  not built yet

bc/ruby-hello-world docker build of https://github.com/openshift/ruby-hello-world
  not built yet

bc/ruby-something-else docker build of https://github.com/openshift/ruby-hello-world
  not built yet

Errors:
  * bc/ruby-22-centos7 is pushing to istag/ruby-hello-world:latest, but the image stream for that tag does not exist.
  * bc/ruby-hello-world is pushing to istag/ruby-something-else:latest, but the image stream for that tag does not exist.
  * bc/ruby-something-else is pushing to istag/ruby-22-centos7:latest, but the image stream for that tag does not exist.
Warnings:
  * Cycle detected in build configurations: bc/ruby-22-centos7 -> istag/ruby-hello-world:latest -> bc/ruby-hello-world -> istag/ruby-something-else:latest -> bc/ruby-something-else -> istag/ruby-22-centos7:latest -> bc/ruby-22-centos7
  * bc/ruby-22-centos7 builds from istag/ruby-22-centos7:latest, but the image stream tag does not exist.
    try: oc start-build bc/ruby-something-else
  * bc/ruby-hello-world builds from istag/ruby-hello-world:latest, but the image stream tag does not exist.
    try: oc start-build bc/ruby-22-centos7
  * bc/ruby-something-else builds from istag/ruby-something-else:latest, but the image stream tag does not exist.
    try: oc start-build bc/ruby-hello-world
  * istag/ruby-hello-world:latest needs to be imported or created by a build.
    try: oc start-build ruby-22-centos7
  * istag/ruby-something-else:latest needs to be imported or created by a build.
    try: oc start-build ruby-hello-world
  * istag/ruby-22-centos7:latest needs to be imported or created by a build.
    try: oc start-build ruby-something-else

View details with 'oc describe <resource>/<name>' or list everything with 'oc get all'.
```